### PR TITLE
[feat] try to reconnect after each disconnection.

### LINKED ISSUE #84: Reconnect to server right after losing connection
Do something in the lines of f4b2672fb08ffd770f1d09bea2efd67a8474fbd6

### PR DESCRIPTION
**Pull request by rulojuka**
Closes #84

### CODE REVIEW

File: src/main/java/br/com/sbk/sbking/networking/kryonet/KryonetClientFactory.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 public class KryonetClientFactory {
   public static KryonetSBKingClient getRegisteredClient(SBKingClient sbkingClient) {
```
Comment (aateg) [inline src/main/java/br/com/sbk/sbking/networking/kryonet/KryonetClientFactory.java:6]: As Sonarcloud pointed out, `sbkingClient` parameter is not necessary anymore

File: src/main/java/br/com/sbk/sbking/networking/kryonet/KryonetClientListenerFactory.java [hunk 2]
```diff
@@ -53,7 +54,22 @@ private void addToQueue(SBKingMessage message) {
...
+        } catch (Exception e) {
+          LOGGER.fatal("client should be a KryonetSBKingClient");
+          LOGGER.fatal(client);
+          return;
+        }
+
+        // This runs in another Thread because kryonet throws IllegalStateException:
+        // "Cannot connect on the connection's update thread."
+        ReconnectToServerRunner reconnectToServerRunner = new ReconnectToServerRunner(kryonetSBKingClient);
+        new Thread(reconnectToServerRunner).start();
```
Comment (aateg) [inline src/main/java/br/com/sbk/sbking/networking/kryonet/KryonetClientListenerFactory.java:72]: Why this thread doesn't have a name?

**Review by aateg — APPROVED** [review]

File: src/main/java/br/com/sbk/sbking/networking/kryonet/KryonetClientFactory.java (hunk 1 above)
Comment (rulojuka) [reply src/main/java/br/com/sbk/sbking/networking/kryonet/KryonetClientFactory.java:6]: Nice, thanks for noticing!

File: src/main/java/br/com/sbk/sbking/networking/kryonet/KryonetClientListenerFactory.java (hunk 2 above)
Comment (rulojuka) [reply src/main/java/br/com/sbk/sbking/networking/kryonet/KryonetClientListenerFactory.java:72]: Good question. calling it "reconnect"

**Pull request merged**